web/Services/LandlordService.cs

```diff
@@ -85,9 +84,19 @@ await using (var transaction = await _dbContext.Database.BeginTransactionAsync(I
         return _dbContext.Landlords.SingleOrDefaultAsync(l => l.Id == id);
     }
 
-    public List<PropertyDbModel> GetListOfProperties(int landlordId)
+    public async Task ApproveLandlord(int landlordId,  BricksAndHeartsUser user)
     {
-        return _dbContext.Properties
-            .Where(p => p.LandlordId == landlordId).ToList();
+        var landlord = _dbContext.Landlords.Single(l =>l.Id == landlordId);
+        if (!landlord.CharterApproved)
+        {
+            landlord.CharterApproved = true;
+            landlord.ApprovalTime = DateTime.Now;
+            landlord.ApprovalAdminId = user.Id;
+            await _dbContext.SaveChangesAsync();
+        }
+        else
+        {
+            throw new Exception("Landlord already approved");
+        }

```

Comment:
Nothing wrong with the logic, but I prefer to shape methods as:

- get info
- if something isn't valid, throw an error
- otherwise continue and do the thing I want to do

This reduces the amount of code blocks. i.e. 

- get landlord
- if already approved, throw error (basically an early return)
- otherwise carry on and approve